semantic-diff/semantic-diff.cabal

```diff
@@ -0,0 +1,72 @@
+cabal-version:       2.4
+
+name:                semantic-diff
+version:             0.0.0.0
+synopsis:            TODO
+description:         TODO
+homepage:            https://github.com/github/semantic/tree/master/semantic-diff#readme
+bug-reports:         https://github.com/github/semantic/issues
+license:             MIT
+license-file:        LICENSE
+author:              The Semantic authors
+maintainer:          opensource+semantic@github.com
+copyright:           (c) 2019 GitHub, Inc.
+category:            Language
+build-type:          Simple
+stability:           alpha
+extra-source-files:  README.md
+
+tested-with:
+  GHC == 8.6.5
+
+common common
+  default-language: Haskell2010
+  ghc-options:
+    -Weverything
+    -Wno-missing-local-signatures
+    -Wno-missing-import-lists
+    -Wno-implicit-prelude
+    -Wno-safe
+    -Wno-unsafe
+    -Wno-name-shadowing
+    -Wno-monomorphism-restriction
+    -Wno-missed-specialisations
+    -Wno-all-missed-specialisations
+    -Wno-star-is-type
+  if (impl(ghc >= 8.8))
+    ghc-options: -Wno-missing-deriving-strategies
+
+library
+  import: common
+  hs-source-dirs: src
+  exposed-modules:   Data.Diff
+                   , Data.Algebra
+                   , Data.Edit
+                   , Data.Functor.Classes.Generic
+                   , Data.JSON.Fields
+                   , Data.Term
+                   , Diffing.Algorithm
+                   , Diffing.Algorithm.RWS
+                   , Diffing.Algorithm.RWS.FeatureVector
+                   , Diffing.Algorithm.SES
+                   , Diffing.Interpreter
+                   , Prologue
+  build-depends:     base >= 4.13 && < 5
+                   , aeson ^>= 1.4.2.0
+                   , recursion-schemes ^>= 5.1
+                   , lens >= 4.17 && < 4.19
+                   , semantic-source ^>= 0.0.1
+                   , fastsum ^>= 0.1.1.1
+                   , text ^>= 1.2.3.1
+                   , containers ^>= 0.6.0.1
+                   , ghc-prim ^>= 0.5.3
+                   , mersenne-random-pure64 ^>= 0.2.2.0
+                   , array ^>= 0.5.3.0
+                   , generic-monoid ^>= 0.1.0.0
+                   , kdt ^>= 0.2.4
+                   , mtl ^>= 2.2.2
+                   , hashable >= 1.2.7 && < 1.4
+                   , fused-effects ^>= 1
+                   , semilattices ^>= 0.0.0.3
+                   , bytestring ^>= 0.10.8.2
+                   , bifunctors ^>= 5.5
```

Comment:
`build-depends` sadly doesn’t accept comma-free formatting, but formatting the actual dependencies to start on the next line & alphabetizing makes it easier to diff & update this, too:

```cabal
build-depends:
    aeson^>= 1.4.2.0
  , base >= 4.13 && < 5
  …
```